### PR TITLE
html_css_popup

### PR DESCRIPTION
**Pull request by Temu4**
[HTLM_CSS_Popup](https://github.com/kottans/frontend/blob/master/tasks/html-css-popup.md)

### HTML & CSS practice: Google-style Popup

[Demo](https://temu4.github.io/Booble-popup/) |
[Code base](https://github.com/Temu4/Booble-popup)

Please, review.

### CODE REVIEW

**Comment by madmaxWMFU** [comment]
Good job!
The caption is very funny and colorful.

File: submissions/Temu4/html_css_popup/index.html [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        <a href="#" tabindex="1">Bmail</a>
+        <!-- Link for images -->
+        <a href="#" tabindex="2">Images</a>
+        <!-- Link for apps -->
+        <a class="icon-apps bar" href="#" title="Приложения Booble" tabindex="3"
+          ><img src="img/icon-apps.png" alt="Icon apps" />
+        </a>
+        <!-- Container with apps list -->
+        <div class="apps-list">
+          <table>
```
Comment (madmaxWMFU) [inline submissions/Temu4/html_css_popup/index.html]: In my opinion use for showing list in <table>  is not good practice.
Try using blocks using the display style `inline-block`
Check the alignment of the content in the center, after the appearance of the scroll, everything did not move to the left

Comment (Temu4) [reply submissions/Temu4/html_css_popup/index.html]: thanks for the advice! I'll try to fix it.

File: submissions/Temu4/html_css_popup/style.css [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+  vertical-align: middle;
+}
+
+/**  Links **/
+a {
+  width: 20%;
+  margin: auto;
+  color: rgb(22, 22, 24);
+  text-align: center;
+  outline: none;
```
Comment (Shramkoweb) [inline submissions/Temu4/html_css_popup/style.css]: All interactive elements should have :focus state

Comment (Shramkoweb) [reply submissions/Temu4/html_css_popup/style.css]: Dont use styling by tag

File: submissions/Temu4/html_css_popup/index.html [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                  <div class="app-item-label">Planet</div>
+                </td>
+                <td tabindex="24">
+                  <img src="img/apps-collection.png " alt="Collections" />
+                  <div class="app-item-label">Collections</div>
+                </td>
+              </tr>
+            </table>
+
+            <div class="apps-even-more-link">
```
Comment (Shramkoweb) [inline submissions/Temu4/html_css_popup/index.html]: `class="apps-even-more-link"`
It's BEM ?

File: submissions/Temu4/html_css_popup/index.html [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+              <a href="#" tabindex="25">Even more from Booble</a>
+            </div>
+          </div>
+        </div>
+        <!-- Link for messages -->
+        <a href="#" class="bar" title="Уведомлений Booble: 0"
+          ><img src="img/icon-bell.png" alt="Icon messages"
```
Comment (Shramkoweb) [inline submissions/Temu4/html_css_popup/index.html]: Good practic use `alt` for `img` 👍

File: submissions/Temu4/html_css_popup/style.css [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+  }
+
+  .search-button {
+    width: 150px;
+  }
+
+  .footer {
+    margin-top: 10%;
+  }
+}
```
Comment (Shramkoweb) [inline submissions/Temu4/html_css_popup/style.css]: Add empty line in the end of every file, or use `editorconfig`

File: submissions/Temu4/html_css_popup/style.css (hunk 2 above)
Comment (A-Ostrovnyy) [reply submissions/Temu4/html_css_popup/style.css]: Also, all interactive elements should have `:active` state.

File: submissions/Temu4/html_css_popup/style.css [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+  width: 70%;
+  cursor: auto;
+}
+/* Apps row */
+tr {
+  width: 100%;
+}
+/* Apps item */
+td {
+  /* width: 33%; */
```
Comment (A-Ostrovnyy) [inline submissions/Temu4/html_css_popup/style.css]: Don't commit commented lines.

File: submissions/Temu4/html_css_popup/index.html (hunk 3 above)
Comment (Temu4) [reply submissions/Temu4/html_css_popup/index.html]: > It's BEM ?

 Sorry, could you explain?

File: submissions/Temu4/html_css_popup/style.css [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+  flex-wrap: nowrap;
+  font-weight: bold;
+  font-size: 60px;
+  padding-left: 0;
+  padding-right: 0;
+  padding-bottom: 40px;
+  -webkit-user-select: none;
+}
+
+.b {
```
Comment (A-Ostrovnyy) [inline submissions/Temu4/html_css_popup/style.css]: Write more meaningful class names.

File: submissions/Temu4/html_css_popup/index.html (hunk 4 above)
Comment (Temu4) [reply submissions/Temu4/html_css_popup/index.html]: 👌

File: submissions/Temu4/html_css_popup/index.html (hunk 1 above)
Comment (Shramkoweb) [reply submissions/Temu4/html_css_popup/index.html]: [link](https://en.bem.info/) Good methodology for naming class.
Recomend for read =)

File: submissions/Temu4/html_css_popup/index.html (hunk 3 above)
Comment (Shramkoweb) [reply submissions/Temu4/html_css_popup/index.html]: [link](https://en.bem.info/) Good methodology for naming class.
Recomend for read =)

**Comment by Temu4** [comment]
Issue: who can help with clicking on popup-menu icon by keyboard??

**Comment by madmaxWMFU** [comment]
There is a [demo](https://jsfiddle.net/0aw4Lf36/) which help for you.
This resource was recommended to me by @kaizengami

File: submissions/Temu4/html_css_popup/index.html [hunk 8]
```diff
@@ -0,0 +1,301 @@
...
+                      <img src="img/apps-blogger.png" alt="Blogger" />
+                    </div>
+                    <div class="app-item__label"><p>Blogger</p></div>
+                  </a>
+                </div>
+                <div class="app-item">
+                  <a class="app-item__link" href="#">
+                    <div class="app-item__icon">
+                      <img src="img/apps-photos.png" alt="Photos" />
+                    </div>
```
Comment (madmaxWMFU) [inline submissions/Temu4/html_css_popup/index.html]: I'd rather use `background` or `background-image` to display icon and don't use contraction `<a><div>`, tag div is superfluous.

File: submissions/Temu4/html_css_popup/index.html [hunk 9]
```diff
@@ -0,0 +1,301 @@
...
+          </div>
+        </li>
+        <!-- Link for messages -->
+        <li class="header__link link">
+          <a
+            href="#"
+            title="Уведомлений Booble: 0"
+            class="link__label"
+            tabindex="0"
+            ><span class="link__content" tabindex="-1"
```
Comment (madmaxWMFU) [inline submissions/Temu4/html_css_popup/index.html]: Do not transfer each attribute to a new line.
When the focus is on the element the tag is highlighted but not the icon.

File: submissions/Temu4/html_css_popup/index.html [hunk 10]
```diff
@@ -0,0 +1,301 @@
...
+                  <a class="app-item__link" href="#">
+                    <div class="app-item__icon">
+                      <img src="img/apps-collection.png " alt="Collections" />
+                    </div>
+                    <div class="app-item__label"><p>Collections</p></div>
+                  </a>
+                </div>
+                <div class="apps-box__even-more even-more">
+                  <a class="even-more__link" href="#">Even more from Booble</a>
+                </div>
```
Comment (madmaxWMFU) [inline submissions/Temu4/html_css_popup/index.html]: Apply `padding` for better readability.

File: submissions/Temu4/html_css_popup/style.css [hunk 11]
```diff
@@ -0,0 +1,349 @@
...
+  opacity: 0.8;
+}
+
+:focus > .link__content {
+  box-shadow: 0 0 3px 1px rgb(0, 140, 255);
+}
+
+/* Removing default outline only after we've added our custom one */
+.link__label:focus,
+.link__label--apps:focus,
```
[the remark this answers is not in the document]
Comment (Temu4) [reply submissions/Temu4/html_css_popup/style.css]: Please, could you explain?

File: submissions/Temu4/html_css_popup/index.html (hunk 9 above)
Comment (Temu4) [reply submissions/Temu4/html_css_popup/index.html]: It's config of my IDE. For me it's useful.

Comment (A-Ostrovnyy) [reply submissions/Temu4/html_css_popup/index.html]: It's easier to read. I do in the same way.

File: submissions/Temu4/html_css_popup/index.html (hunk 10 above)
Comment (Temu4) [reply submissions/Temu4/html_css_popup/index.html]: +

File: submissions/Temu4/html_css_popup/style.css (hunk 11 above)
[the remark this answers is not in the document]
Comment (A-Ostrovnyy) [reply submissions/Temu4/html_css_popup/style.css]: Avoid dumb comments. In the comments should says what you do in this block and why or description of the current block.
Better short comment than no.
In this case, this is a good comment.

File: submissions/Temu4/html_css_popup/index.html (hunk 10 above)
Comment (Temu4) [reply submissions/Temu4/html_css_popup/index.html]: 👌

**Comment by Temu4** [comment]
> Issue: who can help with clicking on popup-menu icon by keyboard??

Finally, understand what is wrong: my checkbox and popup div were from different parents, and I can't manipulate them with :checked status.
Also, regardless of 

> HTML 5 states that the <a> element "may be wrapped around entire paragraphs, lists, tables, and so  forth, even entire sections, so long as there is no interactive content within (e.g. buttons or other links)".
> HTML 4.01 specifies that <a> elements may only contain inline elements. A <div> is a block element, so it may not appear inside an <a>.

when code was rendered by Chrome, div appeared outside `<a>`  tag.

File: submissions/Temu4/html_css_popup/index.html (hunk 8 above)
Comment (Temu4) [reply submissions/Temu4/html_css_popup/index.html]: `<div>` tags allow me to manipulate with elements in `flex` container.

File: submissions/Temu4/html_css_popup/style.css (hunk 11 above)
[the remark this answers is not in the document]
Comment (madmaxWMFU) [reply submissions/Temu4/html_css_popup/style.css]: As I understood, after checking my code, comment there should not be in a major release.
For this it is better to create a new branch in git.

**Comment by Temu4** [comment]
Will it be merged?

**Comment by Temu4** [comment]
> There is a [demo](https://jsfiddle.net/0aw4Lf36/) which help for you.
> This resource was recommended to me by @kaizengami

Useless in my case.
[This](https://www.youtube.com/watch?v=E6kLaaQFctU&t=783s) is really help.

**Comment by A-Ostrovnyy** [comment]
In the demo same way to hide checkbox as in the video).

**Comment by Temu4** [comment]
Oh, I've missed 👓  🙂

**Comment by Temu4** [comment]
fixed bugs

**Comment by A-Ostrovnyy** [comment]
Well done!

**Pull request merged**